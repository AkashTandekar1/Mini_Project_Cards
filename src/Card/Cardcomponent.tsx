import React, { useState } from "react";
import Imagecomponent from "../Image/Imagecomponent.tsx";
import Labelcomponent from "../Label/Labelcomponent.tsx";
import Iconcomponent from "../Icon/Iconcomponent.tsx";
import {
  Container,
  CardContainer,
  LabelCardContainer,
  IconCardContainer,
} from "../Card/CardStyle";

export default function Cardcomponent() {
  const [CardData, SetCardData] = useState<{
    imagedata: string;
    title: string[];
  }>({
    imagedata:
      "https://www.simpleimageresizer.com/_uploads/photos/79b43382/kingfisher-2046453__340_50.jpg",
    title: ["List Item Title", "List Item Title Subtitle"]
  });

  return (
    <div role='root'>
      <Container data-testid="child">
        <CardContainer>
          <Imagecomponent imgdata={CardData} />
        </CardContainer>

        <LabelCardContainer>
          <Labelcomponent labeldata={CardData} />
        </LabelCardContainer>

        <IconCardContainer>
          <Iconcomponent />
        </IconCardContainer>
        
      </Container>
    </div>
  );
}
