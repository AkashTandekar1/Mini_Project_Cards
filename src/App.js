import './App.css';
import Cardcomponent from './Card/Cardcomponent.tsx';


function App() {
  return (
    <div className="App">
           <Cardcomponent className='akash'/>
    </div>
  );
}

export default App;
