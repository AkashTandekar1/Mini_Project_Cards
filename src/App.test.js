import { render, screen } from "@testing-library/react";
import App from "./App";
import React from 'react'
import Cardcomponent from "./Card/Cardcomponent";

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

// test('Render card component in the document', () => {
//   const component = render(<App />);
//   const childcomponent = component.container
//   expec(childcomponent).toBeInTheDocument();
// });

// jest.mock('./Card/Cardcomponent', () => () => <div data-test-id="Cardcomponent" />)

// test('Should render card component', () => {
//   const { getBytestId } = render(<App/>)
//   expect(getBytestId(/Card/Cardcomponent)).toBeInTheDocument()
// }) 


test('Should render card component', () => {
    // const component = render(<App/>)
    // expect(component.contains(<Cardcomponent/>).exists()).toBeInTheDocument();
    // expect(component.firstChild).toMatchSnapshot()
    const { getByRole, getByTestId } = render(<App/>);

    const root = getByRole('root');
    const child = getByTestId('child');

    expect(root).toContainElement(child);
    
    
    
   

}) 


// expect(container.firstChild.classList.contains('foo')).toBe(true)
// expect(container.getElementsByClassName('default').length).toBe(1);

