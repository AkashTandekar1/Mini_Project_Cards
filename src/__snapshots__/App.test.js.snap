// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render card component 1`] = `undefined`;
